packages/core/assistant/src/functions/services.ts

```diff
@@ -5,42 +5,66 @@
 import { AiTool, type AiToolkit } from '@effect/ai';
 import { Context, Effect, Layer, Record, Schema } from 'effect';
 
-import { AiToolNotFoundError, ToolExecutionService, ToolResolverService } from '@dxos/ai';
+import { ToolExecutionService, ToolResolverService } from '@dxos/ai';
 import { todo } from '@dxos/debug';
-import { type FunctionDefinition, LocalFunctionExecutionService } from '@dxos/functions';
+import { Obj, Query } from '@dxos/echo';
+import {
+  DatabaseService,
+  FunctionDefinition,
+  type FunctionImplementationResolver,
+  FunctionInvocationService,
+  FunctionType,
+  getUserFunctionIdInMetadata,
+} from '@dxos/functions';
 import { invariant } from '@dxos/invariant';
+import { log } from '@dxos/log';
 
 export const makeToolResolverFromFunctions = (
```

Comment:
add jsdoc explainig the resolition process

1. toolkit
2. functions in db
3. functions passed in